subscription.go

```diff
@@ -29,37 +28,35 @@ type ISubscription interface {
 }
 
 type Subscription struct {
-	addresses         []common.Address
-	backfillOnce      sync.Once
-	cache             *lru.Cache
-	client            *ethclient.Client
-	context           context.Context
-	connector         *Connector
-	done              chan bool
-	errchan           chan error
-	fromBlock         uint64
-	headers           chan *types.Header
-	interrupt         chan os.Signal
-	isHeaderRequired  bool
+	context context.Context
```

Comment:
[Don't add a Context member to a struct type; instead add a ctx parameter to each method on that type that needs to pass it along. The one exception is for methods whose signature must match an interface in the standard library or in a third party library.](https://github.com/golang/go/wiki/CodeReviewComments#contexts)